static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -150,27 +154,49 @@ function WidgetViewerModal(props: Props) {
                   ? FULL_TABLE_ITEM_LIMIT
                   : HALF_TABLE_ITEM_LIMIT
               }
+              cursor={pagination.cursor}
             >
-              {({transformedResults, loading}) => {
+              {({transformedResults, loading, pageLinks}) => {
                 return (
-                  <GridEditable
-                    isLoading={loading}
-                    data={transformedResults}
-                    columnOrder={columnOrder}
-                    columnSortBy={columnSortBy}
-                    grid={{
-                      renderHeadCell: renderGridHeaderCell({
-                        ...props,
-                      }) as (
-                        column: GridColumnOrder,
-                        columnIndex: number
-                      ) => React.ReactNode,
-                      renderBodyCell: renderGridBodyCell({
-                        ...props,
-                      }),
-                    }}
-                    location={location}
-                  />
+                  <React.Fragment>
+                    <GridEditable
+                      isLoading={loading}
+                      data={transformedResults}
+                      columnOrder={columnOrder}
+                      columnSortBy={columnSortBy}
+                      grid={{
+                        renderHeadCell: renderIssueGridHeaderCell({
+                          ...props,
+                          widget: tableWidget,
+                        }) as (
+                          column: GridColumnOrder,
+                          columnIndex: number
+                        ) => React.ReactNode,
+                        renderBodyCell: renderGridBodyCell({
+                          ...props,
+                          widget: tableWidget,
+                        }),
+                      }}
+                      location={location}
+                    />
+                    <StyledPagination
+                      pageLinks={pageLinks}
+                      onCursor={(nextCursor, _path, _query, delta) => {
+                        let nextPage: number | undefined = isNaN(pagination.page)
```

Comment:
can `nextPage` get also an `undefined` value?

Reply:
Shouldn't be possible to get an `undefined` value here, I've updated the type to reflect this!